msdata-sqlite/compare.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Comparison to System.Data.SQLite - Microsoft.Data.Sqlite
```

Comment:
@mistachkin I'd love your feedback on this page. Any other confusion you see from users between System.Data.SQLite and Microsoft.Data.Sqlite?

Reply:
I'm unable to view the document without signing into a Microsoft account?  I'm not able to do that at the moment.

Reply:
lol, sounds about right. Here's a screenshot:
![image](https://user-images.githubusercontent.com/1226749/70560339-4272db80-1b3d-11ea-97e2-c65a0556220b.png)

Reply:
Read through it quickly.  Looks good so far.

I'm curious how Microsoft.Data.SQLite handles the native library integration.  For System.Data.SQLite, it generally uses the "Native Library Pre-Loading" feature, allowing it to dynamically select the native library that matches the processor architecture of the current process.

I'm also curious about support for other features, such as the online backup API, incremental BLOB I/O, and the session extension.  I assume that Microsoft.Data.SQLite does not yet support these?

Reply:
We use [ericsink/SQLitePCL.raw](https://github.com/ericsink/SQLitePCL.raw) to handle all the native interaction (also used by [praeclarum/sqlite-net](https://github.com/praeclarum/sqlite-net)). It's a mixture of MSBuild, NuGet, and the new [NativeLibrary](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.nativelibrary) functionality added to .NET Core.

We have a non-yielding (blocking, all-at-once) API for online backup (see [this sample](https://github.com/bricelam/EntityFramework.Docs/blob/sqlite/samples/msdata-sqlite/BackupSample/Program.cs)) and issue on the backlog to add an interruptable, progress-reporting async version (issue https://github.com/aspnet/EntityFrameworkCore/issues/13834).

Our BLOB I/O looks mostly the same except we implement IStream to make it more versatile. (see [this sample](https://github.com/bricelam/EntityFramework.Docs/blob/sqlite/samples/msdata-sqlite/StreamingSample/Program.cs))

No support for the session extension. The precompiled versions of SQLite that ship as part of SQLitePCLRaw don't include it.

Reply:
"No support for the session extension. The precompiled versions of SQLite that ship as part of SQLitePCLRaw don't include it."

Should I add this?  I don't think anybody has ever asked for it.

Reply:
Note, almost all of the extra-ADO.NET features (Online backup, UDFs, Custom collation, and BLOB I/O) have been contributed by the community. They're not typically a priority for us the EF Core team.

Reply:
@ericsink Ditto. I haven't heard anyone ask for it.